builtins/amp-video.js

```diff
@@ -49,24 +69,29 @@ export function installVideo(win) {
         assertHttpsUrl(this.element.getAttribute('src'), this.element);
       }
       this.propagateAttributes(
-          ['src', 'controls', 'autoplay', 'muted', 'loop', 'poster'],
-          video);
-      video.width = getLengthNumeral(width);
-      video.height = getLengthNumeral(height);
-      this.applyFillContent(video, true);
+          ['src', 'controls', 'autoplay', 'muted', 'loop'],
+          this.video_);
+
+      if (this.element.hasAttribute('preload')) {
+        this.video_.setAttribute(
+            'preload', this.element.getAttribute('preload'));
+      } else {
+        this.video_.removeAttribute('preload');
+      }
+
       this.getRealChildNodes().forEach(child => {
+        // Skip the video we already added to the element.
+        if (this.video_ === child) {
+          return;
+        }
         if (child.getAttribute && child.getAttribute('src')) {
           assertHttpsUrl(child.getAttribute('src'), child);
         }
-        video.appendChild(child);
+        this.video_.appendChild(child);
       });
-      this.element.appendChild(video);
 
-      /** @private {?HTMLVideoElement} */
-      this.video_ = video;
-      setStyles(video, {visibility: 'hidden'});
-      return loadPromise(video).then(() => {
-        setStyles(video, {visibility: ''});
+      return loadPromise(this.video_).then(() => {
+        setStyles(this.video_, {visibility: ''});
```

Comment:
Let's remove as well since we are not hiding it above. Basically we don't need `then` here.